src/test/java/api/loans/CheckOutCalculateDueDateTests.java

```diff
@@ -736,27 +741,32 @@ private DateTime getStartDateTimeOpeningDayRollover(List<OpeningDayPeriod> openi
             LocalDate nextDate = nextOpeningDay.getDate();
 
             if (nextOpeningDay.getAllDay()) {
-              return toUtcDateTime(nextDate, MIDNIGHT);
+              return atStartOfDay(nextDate, UTC)
+                .withTime(MIDNIGHT);
             } else {
               OpeningHour openingHour = nextOpeningDay.getOpeningHour().get(0);
-              LocalTime startTime = openingHour.getStartTime();
-              return toUtcDateTime(nextDate, startTime);
+
+              return atStartOfDay(nextDate, UTC)
+                .withTime(openingHour.getStartTime());
```

Comment:
Why does it need to be converted to the start of the day before replacing the time component?

```suggestion
              return atStartOfDay(nextDate, UTC).withTime(openingHour.getStartTime());
```

Reply:
> The original test is calling `toUtcDateTime(nextDate, startTime);` which gets gets the time in UTC.

Does that mean that the `atStartOfDay` is purely to get the date time representation into UTC?

> The start of the day must be the start of the UTC day.

Why is the start of the day relevant?

(I do wonder if UTC is an accidental detail here and is only an unfortunate example of time zone. One thing I've learnt from this endeavour is that as soon as timezone became relevant, all of the tests should have not run in UTC and all of the date time code should likely have been aware of time zone, only the serialized representation being UTC)

Reply:
> I am just focusing on converting only what is needed.

That's an understandable approach to this work. Alas, I think it means we miss opportunities to understand and improve things like this.